src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -16,6 +16,36 @@ public Rectangle(Integer width, Integer length, String color) {
         this.color = color;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) o;
+        if (width == null && rectangle.width != null
+                || width != null && !width.equals(rectangle.width)) {
+            return false;
+        }
+        if (length == null && rectangle.length != null
+                || length != null && !length.equals(rectangle.length)) {
+            return false;
+        }
+        return color == null && rectangle.color == null
+               || color != null && color.equals(rectangle.color);
```

Comment:
see checklist: While comparing fields at equals, use single return statement for all fields(e.g. return check1 && check2 && check3;).